R/available-data.R

```diff
@@ -272,141 +464,288 @@ spod_get_latest_v2_file_list <- function(
-    }
-    latest_data_links_xml_path <- spod_get_latest_v2_file_list(data_dir = data_dir)
-  } else {
-    latest_data_links_xml_path <- utils::tail(xml_files_list, 1)
-  }
-
-  # Check if the XML file is 1 day old or older from its name
-  file_date <- stringr::str_extract(latest_data_links_xml_path, "[0-9]{4}-[0-9]{2}-[0-9]{2}")
-  
-  if (file_date < format(Sys.Date(), format = "%Y-%m-%d")) {
-    if (isFALSE(quiet)) {
-      message("File list xml is 1 day old or older, getting latest data links xml")
-    }
-    latest_data_links_xml_path <- spod_get_latest_v2_file_list(data_dir = data_dir)
+  if (use_s3) {
+    files_table <- tryCatch(
+      {
+        spod_available_data_s3(ver = 2, force = force, quiet = quiet)
+      },
+      error = function(e) {
+        message(
+          "S3 fetch failed (",
+          e$message,
+          "); falling back to XML sequence."
+        )
+        read_data_links_memoised(
+          metadata_folder = metadata_folder,
+          data_dir = data_dir,
+          force = force,
+          quiet = quiet,
+          latest_file_function = spod_get_latest_v2_file_list
+        )
+      }
+    )
   } else {
-    if (isFALSE(quiet)) {
-      message("Using existing data links xml: ", latest_data_links_xml_path)
-    }
-  }
-
-  if (length(latest_data_links_xml_path) == 0) {
-    if (isFALSE(quiet)) {
-      message("Getting latest data links xml")
-    }
-    latest_data_links_xml_path <- spod_get_latest_v2_file_list(data_dir = data_dir)
+    files_table <- read_data_links_memoised(
+      metadata_folder = metadata_folder,
+      data_dir = data_dir,
+      force = force,
+      quiet = quiet,
+      latest_file_function = spod_get_latest_v2_file_list
+    )
   }
-  
-  
-  x_xml <- xml2::read_xml(latest_data_links_xml_path)
 
-  files_table <- tibble::tibble(
-    target_url = xml2::xml_find_all(x = x_xml, xpath = "//link") |> xml2::xml_text(),
-    pub_date = xml2::xml_find_all(x = x_xml, xpath = "//pubDate") |> xml2::xml_text()
-  )
-
-  files_table$pub_ts <- lubridate::dmy_hms(files_table$pub_date)
   files_table$file_extension <- tools::file_ext(files_table$target_url)
   files_table <- files_table[files_table$file_extension != "", ]
-  files_table$pub_date <- NULL
 
-  files_table$data_ym <- lubridate::ym(stringr::str_extract(files_table$target_url, "[0-9]{4}-[0-9]{2}"))
-  files_table$data_ymd <- lubridate::ymd(stringr::str_extract(files_table$target_url, "[0-9]{8}"))
+  files_table$data_ym <- lubridate::ym(stringr::str_extract(
+    files_table$target_url,
+    "[0-9]{4}-[0-9]{2}"
+  ))
+  files_table$data_ymd <- lubridate::ymd(stringr::str_extract(
+    files_table$target_url,
+    "[0-9]{8}"
+  ))
   # order by pub_ts
   files_table <- files_table[order(files_table$pub_ts, decreasing = TRUE), ]
   files_table$local_path <- file.path(
     data_dir,
-    stringr::str_replace(files_table$target_url, ".*mitma.es/",
-      spod_subfolder_raw_data_cache(ver = 2))
+    stringr::str_replace(
```

Comment:
OK, for consistency maybe it is worth replacing grepl with `str_detect` if we're using {stringr} anyway?

Reply:
Good point, if it ain't broke don't fix it..